src/common/datavcmn.cpp

```diff
@@ -989,7 +989,8 @@ wxDataViewCustomRendererBase::WXCallRender(wxRect rectCell, wxDC *dc, int state)
     const int align = GetEffectiveAlignment();
 
     const wxSize size = GetSize();
-
+    rectItem.width = size.GetWidth();
+    rectItem.height = size.GetHeight();
```

Comment:
I don't think this is correct, cell rect is different from the content rect and it seems wrong to use `rectItem` with the origin taken from the  former and size from the latter. I hope the author can explain why was this needed.

Reply:
(I've been meddling with this rendering code since I last came across this change and wanted to comment, FWIW anyway).

This is indeed incorrect, it's also explained in a comment just below:

`notice that many existing renderers (e.g. wxDataViewSpinRenderer) return hard-coded size which can be more than they need and if we trusted their GetSize() we'd draw the text out of cell bounds entirely`

Using the cell's origin combined with an item/content's size can (therefore) also be correct though, which is done just below the comment (conditionally, if an item's side fits the cell's).